evap/evaluation/models.py

```diff
@@ -609,6 +611,24 @@ def update_courses(cls):
         logger.info("update_courses finished.")
 
 
+@receiver(post_transition, sender=Course)
+def warmup_cache_on_publish(instance, target, **_kwargs):
+    if target == 'published':
+        from evap.results.tools import collect_results
+        from evap.results.views import warm_up_template_cache
+        collect_results(instance)
+        warm_up_template_cache([instance])
+
+
+@receiver(post_transition, sender=Course)
+def delete_cache_on_unpublish(instance, source, **_kwargs):
+    if source == 'published':
+        from evap.results.tools import get_collect_results_cache_key
+        from evap.results.views import delete_template_cache
+        caches['results'].delete(get_collect_results_cache_key(instance))
+        delete_template_cache(instance)
```

Comment:
We've had bugs due to invalid cache clearing. Do we want to add a test for this?

Reply:
there are asserts for caches being cleared after unpublishing. if you want to add some tests, i won't try to stop you :)

Reply:
well, actually there aren't asserts after deletion. so tests would be even better.